opm/autodiff/ParallelOverlappingILU0.hpp

```diff
@@ -81,6 +150,278 @@ namespace Opm
 {
     namespace detail
     {
+    struct Reorderer
+    {
+        virtual std::size_t operator[](std::size_t i) const = 0;
+    };
+
+    struct NoReorderer : public Reorderer
+    {
+        virtual std::size_t operator[](std::size_t i) const
+        {
+            return i;
+        }
+    };
+
+    struct RealReorderer : public Reorderer
+    {
+        RealReorderer(std::vector<std::size_t> ordering)
+            : ordering_(ordering)
```

Comment:
This is a candidate for `std::move()`.

Reply:
No as ordering might still be used outside later. But there should be no need for a copy.